File: practice/jian_zhi_offer/TreeNode.ts
export class BinaryTreeNode {
    public val: any;
    public left: BinaryTreeNode| null;
    public right: BinaryTreeNode | null;
    constructor(val: any) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

export function preTraverse(root: BinaryTreeNode| null) {
    if(root == null) {
        return;
    }
    console.log(root.val);
    preTraverse(root.left);
    preTraverse(root.right);
}

export function midTraverse(root: BinaryTreeNode | null) {
    if(root == null) {
        return;
    }
    midTraverse(root.left);
    console.log(root.val);
    midTraverse(root.right);
}

export function postTraverse(root: BinaryTreeNode | null) {
    if(root == null) {
        return;
    }
    postTraverse(root.left);
    postTraverse(root.right);
    console.log(root.val);
}

File: practice/jian_zhi_offer/onLeetCode/TreeNode.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) {
        val = x;
    }

    static void preTraverse(TreeNode root) {
        if(root == null) {
            return;
        }
        System.out.print(root.val + ",");
        preTraverse(root.left);
        preTraverse(root.right);
    }

    static void midTraverse(TreeNode root) {
        if(root == null) {
            return;
        }
        midTraverse(root.left);
        System.out.print(root.val + ",");
        midTraverse(root.right);
    }

    static void postTraverse(TreeNode root) {
        if(root == null) {
            return;
        }
        postTraverse(root.left);
        postTraverse(root.right);
        System.out.print(root.val + ",");
    }

}